rtgui/toolpanelcoord.cc

```diff
@@ -336,6 +336,24 @@ void ToolPanelCoordinator::panelChanged (rtengine::ProcEvent event, const Glib::
         }
     }
 
+    int tr = TR_NONE;
+    if (params->coarse.rotate == 90) {
+        tr = TR_R90;
+    } else if (params->coarse.rotate == 180) {
+        tr = TR_R180;
+    } else if (params->coarse.rotate == 270) {
+        tr = TR_R270;
+    }
+
+    // Update "on preview" geometry
+    if (event == rtengine::EvPhotoLoaded || event == rtengine::EvProfileChanged || event == rtengine::EvHistoryBrowsed || event == rtengine::EvCTRotate) {
+        // updating the "on preview" geometry
+        int fw, fh;
+        rtengine::ImageSource *ii = (rtengine::ImageSource*)ipc->getInitialImage();
```

Comment:
Not sure what is going on: Either the return type of `StagedImageProcessor::getInitialImage` is wrong or we can't actually assume that the initial image will always be an instance of `ImageSource`. In any case, we should not use a C-style cast. Since it is down casting, we should probably use `dynamic_cast`, e.g.
````
if (const auto ii = dynamic_cast<rtengine::ImageSource*>(ipc->getInitialImage())) { /* ... */ }
````
but as I said, if we are sure about the type, the return type of the method should probably be changed. (At least the only implementation seems to return an instance of `ImageSource`.)